motion_predict/src/predict_ctrv.cpp

```diff
@@ -122,11 +122,12 @@ CTRV_State CTRVPredict(const CTRV_State& state, const double delta_t)
 
   double v_w = state.v / state.yaw_rate;
   double sin_yaw = sin(state.yaw);
+  double cos_yaw = cos(state.yaw);
   double wT = state.yaw_rate * delta_t;
-
-  next_state.x = v_w * sin(wT + state.yaw) - v_w * sin_yaw + state.x;
-  next_state.y = -v_w * cos(wT + state.yaw) + v_w * sin_yaw + state.y;
-  next_state.yaw = wT + state.yaw;
+  
+  next_state.x = state.x + v_w * (sin(state.yaw + wT) - sin_yaw);
+  next_state.y = state.y + v_w * (cos_yaw - cos(state.yaw + wT));
```

Comment:
Should this be `next_state.y = state.y + v_w * (sin_yaw - cos(state.yaw + wT));` or was the `sin_yaw` a typo in the original line?

If this was a typo and needs to be changed, then the `CTRVPredict` unit test will need to be updated again as well.

Reply:
Yeah this is the main equation that is being fixed at line 129

Reply:
Thanks. It looks like our detailed design document is outdated, but I want to at least document where we are getting these equations from. Since we're trying to get the PR in quickly for the upcoming demo, would you mind putting a link (in the PR description) to a reference for these CTRV equations?

Reply:
I updated the description with links. When I work out the equations, the original paper seems to have had a typo:
http://fusion.isif.org/proceedings/fusion08CD/papers/1569107835.pdf

Reply:
which is confirmed by here as well
https://www.reddit.com/r/ControlTheory/comments/hv2et2/need_help_understanding_ctrv_motion_model_for/

The original equation had a typo where y component should have consisted of cos & cos instead of cos & sin